### ISSUE TITLE
Github repository does not show up on AWS Amplify console dropdown

### ISSUE THREAD

**Issue by keichenblat** [aws-amplify/amplify-hosting#2653]
### Before opening, please confirm:

- [X] I have checked to see if my question is addressed in the [FAQ](https://github.com/aws-amplify/amplify-hosting/blob/master/FAQ.md).
- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-hosting/issues?q=is%3Aissue+).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-hosting/blob/master/CONTRIBUTING.md).
- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

### Region

us-east-1

### Amplify Hosting feature

Not applicable

### Describe the bug

I am trying to add a Vue2 web app repository on a new account, but I do not see the repository in the drop down even after making many commits to it

- The repository was successfully linked to amplify on another account approximately 5 months ago
- The repository's visibility is "internal" (it is part of a GitHub organization)
- I have admin access to the repository
- I do see other repositories in the drop down (including internal ones that belong to the same organization)

### Expected behavior

The repository should appear in the drop down

### Reproduction steps

Unknown

### Additional information

- I have already tried the steps mentioned here: https://github.com/aws-amplify/amplify-hosting/issues/1347#issuecomment-1009299429
- The dropdown shows repositories that were created both before and after the repo in question

**Comment by hloriana** [comment]
Hi @keichenblat thanks for raising this issue. After committing to the repository, did you hit the refresh button and run into this same behavior?

**Comment by hloriana** [comment]
Closing due to no response. Please re-open if this is still an issue. Thanks!

**Comment by keichenblat** [comment]
> Closing due to no response. Please re-open if this is still an issue. Thanks!

Apologies for the delay.
Yes, I have tried committing and hitting the refresh button.
I even re-authorized Amplify, as you've once suggested here: https://github.com/aws-amplify/amplify-hosting/issues/1347#issuecomment-1009299429
(but that did not help either)

I have a HAR file that captures this process. Is there a way for me to send it to you privately? 
(P.S. it weighs 25MB uncompressed / 6MB compressed, so it exceeds the 5MB attachment limit of AWS support center)

**Comment by hloriana** [comment]
Can you try sending the HAR to aws-amplify-customer@amazon.com?

**Comment by keichenblat** [comment]
> Can you try sending the HAR to [aws-amplify-customer@amazon.com](mailto:aws-amplify-customer@amazon.com)?

Sent 👌🏻

**Comment by keichenblat** [comment]
The issue has resolved itself today, at least for the specific repository that I was trying to connect.
I'm not 100% sure if it's related, but I had to reconnect the repository to amplify on the AWS account where the repository was already linked and working for months.

After going through the reconnect repository process on the other account, I finally started seeing the repository in the dropdown menu on all my AWS accounts

**Comment by indiejoseph** [comment]
same here, the dropdown only shown my personal repos, cannot see any repo from org.

**Comment by tipsysquid** [comment]
Hi everyone, I ran into this issue myself and found a resolution. 

**This is my timeline:** 

* I created a new github profile that was given limited (Read Only) access to my repo
* Then I went to Amplify and connected the profile
* First, I was unable to see my repo in the drop down
* I then found this issue #2653 that suggested I make a commit to the repo itself, revoke auth to Amplify from Github, then try again
* After reconnecting github auth to Amplify again, I was still unable to see the repo. 
* I then remembered in a previous time connecting Github Oauth that I needed to make the profile part of the organization of the repo using the "Invite member" to Organization feature in Github
* Once I added the profile as a Member of the repo's Github Org, I then revoked auth token to Amplify again
* I restarted the process with Amplify, and was able to see the repo I was looking for _and_ a Request button for access to the Organization! Just a bit of progress
* I hit Request, the admin of the Org got an email to approve, admin approved, and then I needed to rovoke Auth _again_ with   Amplify.
* Then, after granting permission, I reconnected with Amplify and on the dropdown Amplify sees my repo, but it is grayed out and says ["Amplify Hosting requires you to have admin access to the repository"](https://user-images.githubusercontent.com/18171279/161147691-7988a8dd-8464-40e8-bddf-38cb180b149b.png)
* Finally, I had the Org owner upgrade the new user's profile to be an admin of the repo, rovoked auth for Amplify, and then restarted the process, and I was able to see and connect to the repo without issue. 

**The TLDR:**
These are the characteristics of a github profile that I successfully connected Amplify to a repo with 
* Admin of the repo
* Part of the organization that the repo belongs to 
* Committed at least 1 time in the repo   
* Needed to revoke Amplify auth to github every time an access change occurred
